receive_buffer.go

```diff
@@ -76,16 +73,15 @@ func (buf *receiveBuffer) read(b []byte, deadline time.Time) (totalN int, err er
 		// immediately available.
 		b = b[n:]
 		select {
-		case frame, open := <-buf.in:
+		case frame := <-buf.in:
 			// Read next frame, continue loop
-			if !open && frame == nil {
-				// we've hit the end
-				err = io.EOF
-				buf.ackIfNecessary()
-				return
-			}
 			buf.onFrame(frame)
 			continue
+		case <-buf.closed:
```

Comment:
We now check buf.closed instead of buf.in, since we can't close the latter anymore.